timestamp-api/src/main/java/com/palantir/timestamp/TimestampStoreInvalidator.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.timestamp;
+
+public interface TimestampStoreInvalidator {
+    /**
+     * This method corrupts the data in an underlying timestamp bound store. This is important for the safety of
+     * timestamp store migrations; if one migrates AtlasDB to a different timestamp bound store, then the
+     * original timestamp bound store must not be used again if it is not updated with information about what
+     * timestamps the new store has given out.
+     *
+     * As this method performs corruption, it should only be called *after* the migration of the existing timestamp
```

Comment:
Do we have _any_ way of enforcing this? Sounds like a P0 waiting to happen, although mitigated by the inverse method.